tools/targets/NotSupported.targets

```diff
@@ -48,11 +49,12 @@
       <GenAPIPath Condition="'$(MSBuildRuntimeType)' == 'core'"> "$(DotNetCmd) $(ToolsArtifactsDir)netcoreapp2.1\Microsoft.DotNet.GenAPI.dll"</GenAPIPath>
       <GenAPIPath Condition="'$(MSBuildRuntimeType)' != 'core'"> "$(ToolsArtifactsDir)net472\Microsoft.DotNet.GenAPI.exe"</GenAPIPath>
     </PropertyGroup>
+    <message Condition="'$(BuildTools)' != 'true'" Text="Skip building tools." />
 
-    <Exec Command="$(DotnetBuildCmd)" WorkingDirectory="$(GenAPISrcDir)Microsoft.DotNet.GenAPI\" />
-    <Exec Command="$(GenAPIPath) $(GenAPIArgs)" WorkingDirectory="$(ToolRuntimePath)" />
+    <Exec Command="$(DotnetBuildCmd)" WorkingDirectory="$(GenAPISrcDir)Microsoft.DotNet.GenAPI\" Condition="'$(BuildTools)' == 'true'" />
+    <Exec Command="$(GenAPIPath) $(GenAPIArgs)" WorkingDirectory="$(ToolRuntimePath)" Condition="'$(BuildTools)' == 'true'" />
 
-    <ItemGroup>
+    <ItemGroup Condition="'$(BuildTools)' == 'true'">
```

Comment:
Can this be avoided?

Reply:
Without this condition, it will raise an exception if couldn't find the source file `Microsoft.Data.SqlClient.notsupported.cs`.